packages/server-wallet/src/models/objective.ts

```diff
@@ -1,121 +1,90 @@
-    type: objective.type,
-    participants: [],
-    data,
-  };
 }
 
 export class Objective extends Model {
   readonly objectiveId!: ObjectiveStoredInDB['objectiveId'];
   readonly status!: ObjectiveStoredInDB['status'];
   readonly type!: ObjectiveStoredInDB['type'];
-  readonly targetChannelId!: string;
-  readonly fundingStrategy!: FundingStrategy;
-  readonly ledgerId!: string;
-  readonly jointChannelId!: string;
-  readonly guarantorId!: string;
+  readonly data!: ObjectiveStoredInDB['data'];
 
   static tableName = 'objectives';
   static get idColumn(): string[] {
     return ['objectiveId'];
   }
 
+  static relationMappings = {
+    objectivesChannels: {
+      relation: Model.ManyToManyRelation,
+      modelClass: ObjectiveChannel,
+      join: {
+        from: 'objectives.objectiveId',
+        through: {
+          from: 'objectives_channels.objectiveId',
+          to: 'objectives_channels.channelId',
+        },
+        to: 'channels.channelId',
+      },
+    },
+  };
+
   static async insert(
     objectiveToBeStored: ObjectiveType & {
       status: 'pending' | 'approved' | 'rejected' | 'failed' | 'succeeded';
     },
     tx: TransactionOrKnex
   ): Promise<Objective> {
-    switch (objectiveToBeStored.type) {
-      case 'OpenChannel':
-      case 'CloseChannel':
-        return Objective.query(tx).insert({
-          objectiveId: objectiveId(objectiveToBeStored),
-          status: objectiveToBeStored.status,
-          type: objectiveToBeStored.type,
-          targetChannelId: objectiveToBeStored.data.targetChannelId,
-          fundingStrategy: objectiveToBeStored.data.fundingStrategy,
-        });
-      default:
-        throw Error('[unimplemented] You may only store a CloseChannel or OpenChannel objective');
-    }
-  }
-
-  static async forTargetChannelId(
-    targetChannelId: string,
-    tx: TransactionOrKnex
-  ): Promise<ObjectiveStoredInDB> {
-    const objective = await Objective.query(tx)
-      .select()
-      .first()
-      .where({targetChannelId: targetChannelId});
-    return extract(objective);
-  }
+    const id: string = objectiveId(objectiveToBeStored);
 
-  static async forTargetChannelIds(
-    targetChannelIds: string[],
-    tx: TransactionOrKnex
-  ): Promise<ObjectiveStoredInDB[]> {
-    const objectives = await Objective.query(tx)
-      .select()
-      .whereIn('targetChannelId', targetChannelIds);
+    const objective = await Objective.query(tx).insert({
+      objectiveId: id,
+      status: objectiveToBeStored.status,
+      type: objectiveToBeStored.type,
+      data: objectiveToBeStored.data,
+    });
 
-    return objectives.map(extract);
+    // Associate the objective with any channel that it references
+    // By inserting an ObjectiveChannel row for each channel
+    // Requires objective and channels to exist
+    extractReferencedChannels(objectiveToBeStored).map(
+      async value => await ObjectiveChannel.query(tx).insert({objectiveId: id, channelId: value})
+    );
```

Comment:
These lines capture how we would track relations between objectives and channels.